Failing to boot on fresh install when certain extensions preexist in `$HOME/.config/oni2/extensions`

I'm on the latest `macOS Catalina 10.15.4` trying to run the latest build (download on 2020-05-06):
```
Version: 0.4.0
Date: 5/5/2020, 11:50:56 PM
Commit: c609e7c2
```

Oni2 fails to boot on a fresh (re-)install if one has certain extensions already within `$HOME/.config/oni2/extensions`.

**NB.** _A temporary workaround is to temporarily rename one's existing extensions directory before installing and running a new Oni2 build._

The specific extension on my system that is currently causing the failure to boot is:
```
"name": "phpcs",
"description": "PHP CodeSniffer for Visual Studio Code",
"version": "1.0.5",
"author": "Ioannis Kappas",
"publisher": "ikappas",
"license": "MIT",
"repository": {
  "type": "git",
  "url": "https://github.com/ikappas/vscode-phpcs.git"
}
```

I'm not sure if this is the only extension causing the boot failure, so here is a list of the extensions I'm installing:
```
redhat-developer/vscode-yaml.git
vscode-icons/vscode-icons.git
vuejs/vetur.git
mblode/vscode-twig-language-2.git
ArdenIvanov/svelte-intellisense.git
fivethree-team/vscode-svelte-snippets.git
UnwrittenFun/svelte-vscode.git
ahmadawais/shades-of-purple-vscode.git
mrmlnc/vscode-scss.git
hoovercj/vscode-power-mode.git
ikappas/vscode-phpcs.git
MehediDracula/PHP-Namespace-Resolver.git
bmewburn/vscode-intelephense.git
neild3r/vscode-php-docblocker.git
felixfbecker/vscode-php-debug.git
ChristianKohler/PathIntellisense.git
ChristianKohler/NpmIntellisense.git
eamodio/vscode-gitlens.git
microsoft/vscode-eslint.git
dsznajder/vscode-es7-javascript-react-snippets.git
mastazi/VS-code-drupal.git
tsega/drupal-8-twig-snippets-vs-code.git
dssiqueira/drupal-8-snippets-vs-code.git
tsega/drupal-8-javascript-snippets-vs-code.git
ikappas/vscode-composer.git
```

And here is the full debug output from running `/Applications/Onivim2.app/Contents/MacOS/Oni2 -f --trace --debug` locally:
```
[DEBUG]   +17ms Oni2.Exception : Recording backtraces
[DEBUG]    +0ms Oni2.Exception : Recording backtraces
[INFO]     +0ms Oni2_editor : Starting Onivim 2.0.4.0 (c609e7c2)
[DEBUG]    +0ms Oni2_editor : Calling App.start
[DEBUG]   +55ms Oni2_editor : Init
[INFO]     +0ms Oni2_editor : Startup: Changing folder to: /Users/m/Documents
[DEBUG]    +0ms Revery.Core.Window : Starting window creation...
[INFO]     +0ms Revery.Core.Window : Creating window Oni2 width: 800 height: 600
[INFO]    +15ms Revery.Core.Window : Window created successfully.
[DEBUG]    +0ms Revery.Core.Window : - Id: 1
[DEBUG]    +0ms Revery.Core.Window : - PixelFormat: SDL_PIXELFORMAT_ARGB8888
[DEBUG]    +0ms Revery.Core.Window : Setting window context
[DEBUG]   +16ms Revery.Core.Window : GL setup. Checking GL version...
[DEBUG]    +0ms Revery.Core.Window : Checking GL vendor...
[DEBUG]    +0ms Revery.Core.Window : Checking GL shading language version...
[INFO]     +0ms Revery.Core.Window : OpenGL hardware info:
[INFO]     +0ms Revery.Core.Window :   renderer: AMD Radeon Pro 5500M OpenGL Engine
[INFO]     +0ms Revery.Core.Window :   version: 4.1 ATI-3.8.24
[INFO]     +0ms Revery.Core.Window :   vendor: ATI Technologies Inc.
[INFO]     +0ms Revery.Core.Window :   shadingLanguageVersion: 4.10
[DEBUG]    +0ms Revery.Core.Window : Loading icon from: /Applications/Onivim2.app/Contents/MacOS/logo.png
[DEBUG]    +0ms Revery.Core.Window : Icon loaded successfully.
[DEBUG]    +4ms Revery.Core.Window : Icon set successfully.
[DEBUG]    +0ms Revery.Core.Window : Getting window metrics
[DEBUG]    +0ms Revery.Core.Window : Metrics: DevicePixelRatio: 1.000000 ScaleFactor: 1.000000 Zoom: 1.000000 Raw Dimensions: 800x600px Framebuffer: 800x600px
[INFO]     +0ms Revery.Core.Window : Using vsync: Immediate
[DEBUG]    +1ms Oni2_editor : Initializing setup.
[DEBUG]    +0ms Oni2.Core.Setup : Looking for setup configuration at: /Applications/Onivim2.app/Contents/MacOS/setup.json
[DEBUG]    +0ms Oni2_editor : Startup: Starting StoreThread
[INFO]     +1ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-gruvbox/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-seti/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/go/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-nord/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-monokai/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/reason-vscode/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/python/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/css/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 73 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/css-language-features/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 70 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/typescript-language-features/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-abyss/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/rust/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/java/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/html/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/php/package.nls.json
[INFO]     +1ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-quietlight/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-solarized-dark/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/powershell/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/typescript-basics/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/json/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-tomorrow-night-blue/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-solarized-light/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/cpp/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/shellscript/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/markdown-basics/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-defaults/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/csharp/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-monokai-dimmed/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/laserwave/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/javascript/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/onedark-pro/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-kimbie-dark/package.nls.json
[INFO]     +1ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/ruby/package.nls.json
[INFO]     +1ms Exthost.Extension.Scanner : Loaded 124 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/git/package.nls.json
[INFO]     +1ms Oni2.Store.Utility : Searching for user extensions in: /Users/m/.config/oni2/extensions
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-icons/package.nls.json
[ERROR]    +0ms Exthost.Extension.Scanner : Failed to parse /Users/m/.config/oni2/extensions/vscode-icons/package.json:
	Expected a string, but got [ "ui" ]
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-eslint/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-es7-javascript-react-snippets/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-twig-language-2/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/NpmIntellisense/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-intelephense/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/drupal-8-snippets-vs-code/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/svelte-vscode/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-php-debug/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/drupal-8-javascript-snippets-vs-code/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-scss/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-svelte-snippets/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/shades-of-purple-vscode/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/drupal-8-twig-snippets-vs-code/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-php-docblocker/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/PHP-Namespace-Resolver/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-phpcs/package.nls.json
[ERROR]    +0ms Exthost.Extension.Scanner : Failed to parse /Users/m/.config/oni2/extensions/vscode-phpcs/package.json:
	Expected an object with an attribute "engines", but got
{
  "name": "phpcs",
  "description": "PHP CodeSniffer for Visual Studio Code",
  "version": "1.0.5",
  "author": "Ioannis Kappas",
  "publisher": "ikappas",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/ikappas/vscode-phpcs.git"
  },
  "scripts": {
    "test": "npm-run-all --parallel test:*",
    "test:server": "cd phpcs-server && npm test",
    "test:client": "cd phpcs && npm test",
    "preinstall": "rimraf node_modules",
    "postinstall": "npm-run-all --parallel postinstall:*",
    "postinstall:server": "cd phpcs-server && npm install",
    "postinstall:client": "cd phpcs && npm install",
    "compile": "npm-run-all --serial clean:* compile:*",
    "compile:server":
      "cd phpcs-server && npm run installServer && cd .. && tsc -p phpcs-server/tsconfig.json",
    "compile:client": "tsc -p phpcs/tsconfig.json",
    "watch:server":
      "cd phpcs-server && npm run installServer && cd .. && tsc -w -p phpcs-server/tsconfig.json",
    "watch:client": "tsc -w -p phpcs/tsconfig.json",
    "clean": "npm-run-all --parallel clean:*",
    "clean:server": "cd phpcs-server && npm run clean",
    "clean:client": "cd phpcs && npm run clean"
  },
  "devDependencies": {
    "@types/cross-spawn": "6.0.0",
    "@types/expect": "1.20.2",
    "@types/micromatch": "^3.1.0",
    "@types/mocha": "2.2.48",
    "@types/node": "^8.9.4",
    "@types/semver": "^5.5.0",
    "mocha": "^5.0.1",
    "npm-run-all": "^4.1.2",
    "rimraf": "^2.6.2",
    "ts-node": "^5.0.0",
    "tslint": "^5.9.1",
    "typescript": "^2.7.2",
    "typescript-formatter": "^7.1.0"
  }
}
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-power-mode/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/PathIntellisense/package.nls.json
[INFO]     +2ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-gitlens/package.nls.json
[ERROR]    +2ms Oni2.Exception : Exception Re__Perl.Parse_error:
Raised at file "lib/perl.ml", line 111, characters 10-27
Called from file "lib/perl.ml", line 82, characters 12-19
Called from file "lib/perl.ml", line 80, characters 18-26
Called from file "lib/perl.ml" (inlined), line 74, characters 30-41
Called from file "lib/perl.ml", line 252, characters 12-21
Called from file "lib/perl.ml", line 262, characters 4-142
Called from file "src/Core/WhenExpr/WhenExpr.re", line 60, characters 8-74
Called from file "src/Core/WhenExpr/WhenExpr.re", line 112, characters 27-55
Called from file "list.ml", line 92, characters 20-23
Called from file "list.ml", line 92, characters 32-39
Called from file "src/Core/WhenExpr/WhenExpr.re", line 127, characters 15-62
Called from file "list.ml", line 92, characters 20-23
Called from file "src/Core/WhenExpr/WhenExpr.re", line 133, characters 14-61
Called from file "src/decoders_util.ml", line 7, characters 17-22
Called from file "src/Core/Json.re", line 110, characters 21-45
Called from file "src/Exthost/Extension/Contributions.re", line 65, characters 12-147
Called from file "src/Core/Json.re", line 161, characters 15-27
Called from file "src/decode.ml", line 287, characters 16-29
Called from file "src/decoders_util.ml", line 70, characters 17-23
Called from file "src/decoders_util.ml", line 54, characters 17-21
Called from file "src/decode.ml", line 285, characters 12-203
Called from file "src/decode.ml", line 457, characters 18-45
Called from file "list.ml", line 92, characters 20-23
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "src/decode.ml", line 453, characters 12-269
Called from file "src/decode.ml", line 188, characters 42-59
Called from file "src/Core/Json.re", line 110, characters 21-45
Called from file "src/Exthost/Extension/Contributions.re", line 280, characters 15-64
Called from file "src/Core/Json.re", line 161, characters 15-27
Called from file "src/Core/Json.re", line 110, characters 21-45
Called from file "src/Exthost/Extension/Manifest.re", line 81, characters 12-146
Called from file "src/Core/Json.re", line 161, characters 15-27
Called from file "src/decode.ml" (inlined), line 489, characters 4-21
Called from file "src/Exthost/Extension/Scanner.re", line 53, characters 9-58
Called from file "list.ml", line 92, characters 20-23
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "src/Exthost/Extension/Scanner.re", line 80, characters 2-241
Called from file "option.ml", line 24, characters 57-62
Called from file "src/Store/Utility.re", line 26, characters 2-244
Called from file "src/Store/StoreThread.re", line 41, characters 10-61
Called from file "src/Timber.re", line 58, characters 14-17
Called from file "src/Store/StoreThread.re", line 25, characters 6-780
Called from file "src/Store/StoreThread.re", line 104, characters 4-103
Called from file "src/bin_editor/Oni2_editor.re", line 243, characters 6-728
Called from file "src/Core/App.re", line 176, characters 17-34
Called from file "src/bin_editor/Oni2_editor.re", line 321, characters 2-18
```

Culprit seems to be the GitLens extension:
```
[DEBUG]   +17ms Oni2.Exception : Recording backtraces
[DEBUG]    +0ms Oni2.Exception : Recording backtraces
[INFO]     +0ms Oni2_editor : Starting Onivim 2.0.4.0 (c609e7c2)
[DEBUG]    +0ms Oni2_editor : Calling App.start
[DEBUG]   +56ms Oni2_editor : Init
[INFO]     +0ms Oni2_editor : Startup: Changing folder to: /Users/m/Documents
[DEBUG]    +0ms Revery.Core.Window : Starting window creation...
[INFO]     +0ms Revery.Core.Window : Creating window Oni2 width: 800 height: 600
[INFO]    +15ms Revery.Core.Window : Window created successfully.
[DEBUG]    +0ms Revery.Core.Window : - Id: 1
[DEBUG]    +0ms Revery.Core.Window : - PixelFormat: SDL_PIXELFORMAT_ARGB8888
[DEBUG]    +0ms Revery.Core.Window : Setting window context
[DEBUG]   +16ms Revery.Core.Window : GL setup. Checking GL version...
[DEBUG]    +0ms Revery.Core.Window : Checking GL vendor...
[DEBUG]    +0ms Revery.Core.Window : Checking GL shading language version...
[INFO]     +0ms Revery.Core.Window : OpenGL hardware info:
[INFO]     +0ms Revery.Core.Window :   renderer: AMD Radeon Pro 5500M OpenGL Engine
[INFO]     +0ms Revery.Core.Window :   version: 4.1 ATI-3.8.24
[INFO]     +0ms Revery.Core.Window :   vendor: ATI Technologies Inc.
[INFO]     +0ms Revery.Core.Window :   shadingLanguageVersion: 4.10
[DEBUG]    +0ms Revery.Core.Window : Loading icon from: /Applications/Onivim2.app/Contents/MacOS/logo.png
[DEBUG]    +0ms Revery.Core.Window : Icon loaded successfully.
[DEBUG]    +4ms Revery.Core.Window : Icon set successfully.
[DEBUG]    +0ms Revery.Core.Window : Getting window metrics
[DEBUG]    +0ms Revery.Core.Window : Metrics: DevicePixelRatio: 1.000000 ScaleFactor: 1.000000 Zoom: 1.000000 Raw Dimensions: 800x600px Framebuffer: 800x600px
[INFO]     +0ms Revery.Core.Window : Using vsync: Immediate
[DEBUG]    +1ms Oni2_editor : Initializing setup.
[DEBUG]    +0ms Oni2.Core.Setup : Looking for setup configuration at: /Applications/Onivim2.app/Contents/MacOS/setup.json
[DEBUG]    +0ms Oni2_editor : Startup: Starting StoreThread
[INFO]     +1ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-gruvbox/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-seti/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/go/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-nord/package.nls.json
[INFO]     +4ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-monokai/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/reason-vscode/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/python/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/css/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 73 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/css-language-features/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 70 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/typescript-language-features/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-abyss/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/rust/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/java/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/html/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/php/package.nls.json
[INFO]     +1ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-quietlight/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-solarized-dark/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/powershell/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/typescript-basics/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/json/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-tomorrow-night-blue/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-solarized-light/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/cpp/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/shellscript/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/markdown-basics/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-defaults/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/csharp/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-monokai-dimmed/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/laserwave/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/javascript/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/onedark-pro/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/theme-kimbie-dark/package.nls.json
[INFO]     +1ms Exthost.Extension.Scanner : Loaded 2 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/ruby/package.nls.json
[INFO]     +1ms Exthost.Extension.Scanner : Loaded 124 localizations from /Applications/Onivim2.app/Contents/Resources/extensions/git/package.nls.json
[INFO]     +2ms Oni2.Store.Utility : Searching for user extensions in: /Users/m/.config/oni2/extensions
[INFO]     +1ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-eslint/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-es7-javascript-react-snippets/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-twig-language-2/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/NpmIntellisense/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-intelephense/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/drupal-8-snippets-vs-code/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/svelte-vscode/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-php-debug/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/drupal-8-javascript-snippets-vs-code/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-scss/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-svelte-snippets/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/shades-of-purple-vscode/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/drupal-8-twig-snippets-vs-code/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-php-docblocker/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/PHP-Namespace-Resolver/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-power-mode/package.nls.json
[INFO]     +0ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/PathIntellisense/package.nls.json
[INFO]     +2ms Exthost.Extension.Scanner : Loaded 0 localizations from /Users/m/.config/oni2/extensions/vscode-gitlens/package.nls.json
[ERROR]    +1ms Oni2.Exception : Exception Re__Perl.Parse_error:
Raised at file "lib/perl.ml", line 111, characters 10-27
Called from file "lib/perl.ml", line 82, characters 12-19
Called from file "lib/perl.ml", line 80, characters 18-26
Called from file "lib/perl.ml" (inlined), line 74, characters 30-41
Called from file "lib/perl.ml", line 252, characters 12-21
Called from file "lib/perl.ml", line 262, characters 4-142
Called from file "src/Core/WhenExpr/WhenExpr.re", line 60, characters 8-74
Called from file "src/Core/WhenExpr/WhenExpr.re", line 112, characters 27-55
Called from file "list.ml", line 92, characters 20-23
Called from file "list.ml", line 92, characters 32-39
Called from file "src/Core/WhenExpr/WhenExpr.re", line 127, characters 15-62
Called from file "list.ml", line 92, characters 20-23
Called from file "src/Core/WhenExpr/WhenExpr.re", line 133, characters 14-61
Called from file "src/decoders_util.ml", line 7, characters 17-22
Called from file "src/Core/Json.re", line 110, characters 21-45
Called from file "src/Exthost/Extension/Contributions.re", line 65, characters 12-147
Called from file "src/Core/Json.re", line 161, characters 15-27
Called from file "src/decode.ml", line 287, characters 16-29
Called from file "src/decoders_util.ml", line 70, characters 17-23
Called from file "src/decoders_util.ml", line 54, characters 17-21
Called from file "src/decode.ml", line 285, characters 12-203
Called from file "src/decode.ml", line 457, characters 18-45
Called from file "list.ml", line 92, characters 20-23
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "src/decode.ml", line 453, characters 12-269
Called from file "src/decode.ml", line 188, characters 42-59
Called from file "src/Core/Json.re", line 110, characters 21-45
Called from file "src/Exthost/Extension/Contributions.re", line 280, characters 15-64
Called from file "src/Core/Json.re", line 161, characters 15-27
Called from file "src/Core/Json.re", line 110, characters 21-45
Called from file "src/Exthost/Extension/Manifest.re", line 81, characters 12-146
Called from file "src/Core/Json.re", line 161, characters 15-27
Called from file "src/decode.ml" (inlined), line 489, characters 4-21
Called from file "src/Exthost/Extension/Scanner.re", line 53, characters 9-58
Called from file "list.ml", line 92, characters 20-23
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "list.ml", line 92, characters 32-39
Called from file "src/Exthost/Extension/Scanner.re", line 80, characters 2-241
Called from file "option.ml", line 24, characters 57-62
Called from file "src/Store/Utility.re", line 26, characters 2-244
Called from file "src/Store/StoreThread.re", line 41, characters 10-61
Called from file "src/Timber.re", line 58, characters 14-17
Called from file "src/Store/StoreThread.re", line 25, characters 6-780
Called from file "src/Store/StoreThread.re", line 104, characters 4-103
Called from file "src/bin_editor/Oni2_editor.re", line 243, characters 6-728
Called from file "src/Core/App.re", line 176, characters 17-34
Called from file "src/bin_editor/Oni2_editor.re", line 321, characters 2-18
``

--- a/src/Core/WhenExpr/WhenExpr.re
+++ b/src/Core/WhenExpr/WhenExpr.re
@@ -7,7 +7,7 @@ type t =
   | Defined(string)
   | Eq(string, Value.t)
   | Neq(string, Value.t)
-  | Regex(string, option(Re.re))
+  | Regex(string, option([@opaque] Oniguruma.OnigRegExp.t))
   | And(list(t))
   | Or(list(t))
   | Not(t)
@@ -20,7 +20,8 @@ let evaluate = (expr, getValue) => {
     | Eq(key, value) => getValue(key) == value
     | Neq(key, value) => getValue(key) != value
     | Regex(_, None) => false
-    | Regex(key, Some(re)) => Re.execp(re, key |> getValue |> Value.asString)
+    | Regex(key, Some(re)) =>
+      Oniguruma.OnigRegExp.test(key |> getValue |> Value.asString, re)
     | And(exprs) => List.for_all(eval, exprs)
     | Or(exprs) => List.exists(eval, exprs)
     | Not(expr) => !eval(expr)
@@ -59,11 +60,11 @@ module Parse = {
 
       | (Some(start), Some(stop)) =>
         let pattern = String.sub(str, start + 1, stop - start - 1);
-        switch (Re.Pcre.re(pattern) |> Re.compile) {
-        | regex => Some(regex)
-        | exception Re.Perl.Parse_error when strict =>
+        switch (Oniguruma.OnigRegExp.create(pattern)) {
+        | Ok(regex) => Some(regex)
+        | Error(str) when strict =>
           failwith("invalid regexp '" ++ str ++ "', ")
-        | exception Re.Perl.Parse_error => None
+        | Error(_) => None
         };
 
       | (None, None) when strict =>

--- a/src/Core/WhenExpr/WhenExpr.rei
+++ b/src/Core/WhenExpr/WhenExpr.rei
@@ -42,7 +42,7 @@ type t =
   | Defined(string)
   | Eq(string, Value.t)
   | Neq(string, Value.t)
-  | Regex(string, option(Re.re))
+  | Regex(string, option([@opaque] Oniguruma.OnigRegExp.t))
   | And(list(t))
   | Or(list(t))
   | Not(t)

--- a/src/Core/WhenExpr/dune
+++ b/src/Core/WhenExpr/dune
@@ -2,4 +2,4 @@
     (name WhenExpr)
     (public_name Oni2.core.whenExpr)
     (preprocess (pps ppx_deriving.show))
-    (libraries Oni2.core.kernel re timber))
+    (libraries Oni2.core.kernel re timber oniguruma))

--- a/test/Core/WhenExpr/WhenExprTests.re
+++ b/test/Core/WhenExpr/WhenExprTests.re
@@ -5,23 +5,32 @@ let re = (pattern, str) => {
   Re.execp(re, str);
 };
 
+let createContext = pairs => {
+  let context = Hashtbl.create(List.length(pairs));
+  List.iter(((k, v)) => Hashtbl.add(context, k, v), pairs);
+
+  name =>
+    switch (Hashtbl.find_opt(context, name)) {
+    | Some(value) => value
+    | None => WhenExpr.Value.False
+    };
+};
+
 describe("WhenExpr", ({describe, _}) => {
   describe("vscode tests", ({describe, _}) => {
     // Translkated from https://github.com/microsoft/vscode/blob/7fc5d9150569247b3494eb3715a078bf7f8e9272/src/vs/platform/contextkey/test/common/contextkey.test.ts
     // TODO: equals
     // TODO: normalize
     describe("evaluate", ({test, _}) => {
-      let context = Hashtbl.create(4);
-      Hashtbl.add(context, "a", WhenExpr.Value.True);
-      Hashtbl.add(context, "b", WhenExpr.Value.False);
-      Hashtbl.add(context, "c", WhenExpr.Value.String("5"));
-      Hashtbl.add(context, "d", WhenExpr.Value.String("d"));
-
-      let getValue = name =>
-        switch (Hashtbl.find_opt(context, name)) {
-        | Some(value) => value
-        | None => WhenExpr.Value.False
-        };
+      let getValue =
+        createContext(
+          WhenExpr.Value.[
+            ("a", True),
+            ("b", False),
+            ("c", String("5")),
+            ("d", String("d")),
+          ],
+        );
 
       let testExpression = (expr, expected) =>
         test(
@@ -65,23 +74,17 @@ describe("WhenExpr", ({describe, _}) => {
 
   describe("regex", ({test, _}) => {
     test("1729-gitlens-regression", ({expect, _}) => {
-      let context = Hashtbl.create(2);
-      Hashtbl.add(
-        context,
-        "gitlens:activeFileStatus",
-        WhenExpr.Value.String("some revision or other"),
-      );
-      Hashtbl.add(context, "resourceScheme", WhenExpr.Value.String("file$"));
-
-      let getValue = name =>
-        switch (Hashtbl.find_opt(context, name)) {
-        | Some(value) => value
-        | None => WhenExpr.Value.False
-        };
+      let getValue =
+        createContext(
+          WhenExpr.Value.[
+            ("gitlens:activeFileStatus", String("some revision or other")),
+            ("resourceScheme", String("file$")),
+          ],
+        );
 
       let expr = "gitlens:activeFileStatus =~ /revision/ && resourceScheme =~ /^(?!(file|git)$).*$/";
       let rules = WhenExpr.parse(expr);
-      Console.log(WhenExpr.show(rules));
+      //Console.log(WhenExpr.show(rules));
       expect.bool(WhenExpr.evaluate(rules, getValue)).toBe(true);
     })
   });

--- a/test/Core/WhenExpr/WhenExprTests.re
+++ b/test/Core/WhenExpr/WhenExprTests.re
@@ -61,5 +61,28 @@ describe("WhenExpr", ({describe, _}) => {
       testExpression("a && b", true && false);
       testExpression("a && !b && c == 5", true && !false && "5" == "5");
     })
-  })
+  });
+
+  describe("regex", ({test, _}) => {
+    test("1729-gitlens-regression", ({expect, _}) => {
+      let context = Hashtbl.create(2);
+      Hashtbl.add(
+        context,
+        "gitlens:activeFileStatus",
+        WhenExpr.Value.String("some revision or other"),
+      );
+      Hashtbl.add(context, "resourceScheme", WhenExpr.Value.String("file$"));
+
+      let getValue = name =>
+        switch (Hashtbl.find_opt(context, name)) {
+        | Some(value) => value
+        | None => WhenExpr.Value.False
+        };
+
+      let expr = "gitlens:activeFileStatus =~ /revision/ && resourceScheme =~ /^(?!(file|git)$).*$/";
+      let rules = WhenExpr.parse(expr);
+      Console.log(WhenExpr.show(rules));
+      expect.bool(WhenExpr.evaluate(rules, getValue)).toBe(true);
+    })
+  });
 });

